eahub/localgroups/views.py

```diff
@@ -1,14 +1,37 @@
 import json
-from datetime import datetime 
+from datetime import datetime
 
 from django.urls import reverse_lazy
 from django.shortcuts import render, redirect, get_object_or_404
 from django.contrib.auth.decorators import login_required
-
+from django.core.mail import send_mail
+from django.contrib.sites.shortcuts import get_current_site
+from .forms import ClaimGroupForm
 from .models import LocalGroup as Group
 
 def view_group(request, slug):
     group = get_object_or_404(Group, slug=slug)
+
     return render(request, 'eahub/group.html', {
         'group': group
     })
+
+def claim_group(request, slug):
+    group = get_object_or_404(Group, slug=slug)
+    if request.method == 'POST':
+        if request.user.is_anonymous:
+            return render(request, 'eahub/claim_group_error.html')
+        else:
+            subject = "EA Group claimed: {0}".format(group.name)
+            user_eahub_url = "https://{0}/profile/{1}".format(get_current_site(request).domain,request.user.profile.slug)
+            message = """ User <a href="{0}">{1}</a> wants to claim group {2}.
+                          Please check whether the user is eligible to claim this group
+                          and let them know of the outcome by emailing them at {3}.
+                      """.format(user_eahub_url, request.user.profile.name, group.name, request.user.email)
+            send_mail(subject, message, request.user.email, ['admin@eahub.org'])
+            return render(request, 'eahub/claim_group_success.html')
+    else:
+        form = ClaimGroupForm()
+        return render(request, 'eahub/claim_group.html', {
```

Comment:
Similarly, I'm not convinced we need this as a separate page, as opposed to just a button on the group page.

Reply:
added a button, which displays textfield with more info on same page